pvlib/temperature.py

```diff
@@ -978,3 +978,397 @@ def prilliman(temp_cell, wind_speed, unit_mass=11.1, coefficients=None):
     return smoothed
+
+
+def generic_linear(poa_global, temp_air, wind_speed, u_const, du_wind,
+                   module_efficiency, absorptance):
+    """
+    Calculate cell temperature using a generic linear heat loss factor model
+    as developed in [1]_.
+
+    The parameters for this model can be obtained using
+    :py:class:`GenericLinearModel`
+
+    Parameters
+    ----------
+    poa_global : numeric
+        Total incident irradiance [W/m^2].
+
+    temp_air : numeric
+        Ambient dry bulb temperature [C].
+
+    wind_speed : numeric
+        Wind speed at a height of 10 meters [m/s].
+
+    u_const : numeric
+        Combined heat transfer coefficient at zero wind speed [(W/m^2)/C]
+
+    du_wind : numeric
+        Influence of wind speed on combined heat transfer coefficient
+        [(W/m^2)/C/(m/s)]
+
+    module_efficiency : float
+        The electrical efficiency of the module. [-]
+
+    absorptance : float
+        The light absorptance of the module. [-]
+
+    Returns
+    -------
+    numeric, values in degrees C.
+
+    References
+    ----------
+    .. [1] A. Driesse et al, "PV Module Operating Temperature
+       Model Equivalence and Parameter Translation". 2022 IEEE
+       Photovoltaic Specialists Conference (PVSC), 2022.
+
+    See also
+    --------
+    pvlib.temperature.GenericLinearModel
+    """
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs, Sept. 2022
+
+    heat_input = poa_global * (absorptance - module_efficiency)
+    total_loss_factor = u_const + du_wind * wind_speed
+    temp_difference = heat_input / total_loss_factor
+
+    return temp_air + temp_difference
+
+
+class GenericLinearModel():
+    '''
+    A generic linear temperature model class that can both use and convert
+    parameters of several well-known equivalent or nearly equivalent models:
+    faiman, pvsyst, noct-sam and sapm.
+
+    Normally this class is used on an ad hoc basis to make a needed conversion
+    to a target model, and the converted parameters are then used for
+    subsequent simulations running the target model.  The parameters are
+    always returned as dictionaries that are compatible with the respective
+    model functions in pvlib.temperature.
+
+    An instance of the class represents a specific module type, therefore,
+    the module properties *efficiency* and *absorptance* must be provided.
+    Although some temperature models do not use these properties, they
+    nevertheless exist and affect operating temperature.
+
+    Since these are empirical models, their parameters are determined
+    from a limited set of measurements (usually high irradiance).
+    The values of efficiency and absorptance should be interpreted
+    as representative of the efficiency and absorptance actually
+    occuring during those measurements.
+
+    When used for time series simulation, efficiency and absorptance may
+    vary, and each empirical model handles this differently.
```

Comment:
```suggestion
    A class that can both use and convert
    parameters of linear module temperature models:
    faiman, pvsyst, noct_sam, sapm_module and general_linear.
    
    Parameters are converted between models by first converting
    to the generic linear heat transfer model [1]_ by the `"use_"`
    methods. The equivalent parameters for the target temperature
    model are then determined by the `"to_"` method. 
    Parameters returned as a dictionary that is compatible with the
    target model.

    An instance of the class represents a specific module type. 
    The parameters ``module_efficiency`` and ``absorptance`` are required.
    Although some temperature models do not use these properties, they
    nevertheless exist and affect operating temperature. Values
    should be representative of the conditions at which the input
    model parameters were determined (usually high irradiance).
```

A suggested rewrite from a "how do I use this" perspective. I don't think we should imply that `module_efficiency` can be a Series even if that might not cause a particular temperature model's code to fail.

Reply:
I took most of your suggested text, but I would say that it should be (or become) standard practice to have module efficiency varying during simulation.